optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/bavet/bi/BavetScoringBiConstraintStream.java

```diff
@@ -113,6 +113,9 @@ private BavetScoringBiConstraintStream(BavetConstraintFactory<Solution_> constra
             if (intMatchWeigher != null) {
                 scoreImpacter = (A a, B b, Consumer<Score<?>> matchScoreConsumer) -> {
                     int matchWeight = intMatchWeigher.applyAsInt(a, b);
+                    if (matchWeight == 0) { // No need to include and no need to undo later.
+                        return () -> { /* NOOP */ };
```

Comment:
Soft: might be worth extracting a NO_OP_UNDO into AbstractConstraintStream?

Reply:
I still wonder if it shouldn't fail fast if the matchWeight is zero.

Actually, thinking about this more - we probably want to have the indictment anyway, even if the matchWeigh is zero...  even if it's just for debugging purposes - for easy of debugging. Let's discuss.